subsys/bluetooth/host/gatt.c

```diff
@@ -2110,47 +2078,74 @@ static bool gatt_cf_notify_multi(struct bt_conn *conn)
 	return CF_NOTIFY_MULTI(cfg);
 }
 
-static int gatt_notify_mult(struct bt_conn *conn, uint16_t handle,
+static int gatt_notify_mult(struct bt_conn *conn, uint16_t num_params,
 			    struct bt_gatt_notify_params *params)
 {
-	struct net_buf **buf = &nfy_mult[bt_conn_index(conn)];
+	struct net_buf *buf;
 	struct bt_att_notify_mult *nfy;
+	struct notify_data data;
+	uint16_t pdu_len = 0;
+
+	for (uint16_t i = 0; i < num_params; i++) {
+		data.attr = params[i].attr;
+		data.handle = bt_gatt_attr_get_handle(params[i].attr);
+		/* Lookup UUID if it was given */
+		if (params[i].uuid) {
+			if (!gatt_find_by_uuid(&data, params[i].uuid)) {
+				return -ENOENT;
+			}
+			params[i].attr = data.attr;
+		} else {
+			if (!data.handle) {
+				return -ENOENT;
+			}
+		}
 
-	/* Check if we can fit more data into it, in case it doesn't fit send
-	 * the existing buffer and proceed to create a new one
-	 */
-	if (*buf && ((net_buf_tailroom(*buf) < sizeof(*nfy) + params->len) ||
-	    !bt_att_tx_meta_data_match(*buf, params->func, params->user_data))) {
-		int ret;
-
-		ret = gatt_notify_mult_send(conn, buf);
-		if (ret < 0) {
-			return ret;
+		/* Confirm that the connection has the correct level of security */
+		if (bt_gatt_check_perm(conn, params[i].attr,
+				       BT_GATT_PERM_READ_ENCRYPT | BT_GATT_PERM_READ_AUTHEN)) {
+			BT_WARN("Link is not encrypted");
+			return -EPERM;
 		}
-	}
 
-	if (!*buf) {
-		*buf = bt_att_create_pdu(conn, BT_ATT_OP_NOTIFY_MULT,
-					 sizeof(*nfy) + params->len);
-		if (!*buf) {
-			return -ENOMEM;
+		/* Calculate the ATT_MULTIPLE_HANDLE_VALUE_NTF pdu length */
+		pdu_len += sizeof(*nfy) + params[i].len;
+		if (pdu_len > UINT16_MAX) {
```

Comment:
This check cannot fail if `pdu_len` is a `uint16_t`. Please change `pdu_len` to `size_t`

Reply:
There is also `u16_add_overflow`.